tests/integration_tests/bugs/test_lp1813396.py

```diff
@@ -0,0 +1,33 @@
+"""Integration test for lp-1813396
+
+Ensure gpg works even if VM provides no /dev/tty"""
```

Comment:
This isn't what we're doing: we're testing that cloud-init passes `--no-tty` to gpg. I think that's sufficient for validation on Ubuntu (particularly as lxd doesn't support _not_ providing `/dev/tty` to a system: https://lxd.readthedocs.io/en/latest/instances/#devices-configuration), but there _could_ be an integration test for this bug which masks/removes `/dev/tty` somehow and then tests that we successfully configure the repo.